src/hidimstat/clustered_inference.py

```diff
@@ -273,20 +281,26 @@ def clustered_inference(
     train_index = _subsampling(n_samples, train_size, groups=groups, seed=seed)
 
     # Clustering
-    X, ward = memory.cache(_ward_clustering)(X_init, ward, train_index)
+    X, ward = _ward_clustering(X_init, ward, train_index)
```

Comment:
```suggestion
    X, ward = memory.cache(_ward_clustering)(X_init, ward, train_index)
```
Note: this takes some time on big image. rather not redo it if it can be avoided !